### ISSUE TITLE
New androidXAnnotation version breaks Android builds

### ISSUE THREAD

**Issue by IgorVanian** [proyecto26/react-native-inappbrowser#398]
### Which platform(s) does your issue occur on?
- Android (on build)

## TL;DR
This line currently installs version 1.6.0 of [annotations](https://developer.android.com/jetpack/androidx/releases/annotation?hl=fr#1.6.0-alpha01)
https://github.com/proyecto26/react-native-inappbrowser/blob/e028173df5ffea7b8c1f3214669cdde33981ee14/android/build.gradle#L67

This version has a `kotlin-stdlib:1.8.0` dependency. By default gradle is using the higher version of this lib in the project, which makes [this issue](https://youtrack.jetbrains.com/issue/KT-55297/kotlin-stdlib-should-declare-constraints-on-kotlin-stdlib-jdk8-and-kotlin-stdlib-jdk7) to pop up.

## Long version
Hello. 
After a lot of debugging, I finally found why current Android builds fail with the following error:
```
[09:58:03]: ▸ > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
[09:58:03]: ▸ > Duplicate class kotlin.collections.jdk8.CollectionsJDK8Kt found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
```

This is due to the merge of jdk7 & jdk8 artifacts that occured in `kotlin-stdlib:1.8.0`. This results in an error if you use packages that have `kotlin-stdlib-jdk7/8 < 1.8.0`. 

From `gradle dependencies` I saw that this package is requiring the latest version of annotations:
<img width="400" alt="image" src="https://user-images.githubusercontent.com/16898357/214889079-5c7053a5-d1e4-4e40-ba95-ba9c1367bbe9.png">
Which can be found here:

https://github.com/proyecto26/react-native-inappbrowser/blob/e028173df5ffea7b8c1f3214669cdde33981ee14/android/build.gradle#L61

The latest 1.6.0 version includes a kotlin 1.8.0 dependency that enables [this issue](https://youtrack.jetbrains.com/issue/KT-55297/kotlin-stdlib-should-declare-constraints-on-kotlin-stdlib-jdk8-and-kotlin-stdlib-jdk7) and makes the build fail.

To fix this issue for the moment, we had to specify a fixed `kotlinVersion=1.8.0` in the `build.gradle` file.

**Comment by balasasidhar** [comment]
+1

**Comment by shuklaaalind** [comment]
How can we resolve this for now?

**Comment by IgorVanian** [comment]
@shuklaaalind I opened a PR for this. But for the moment you can add a custom kotlin version to your project like so:

```
buildscript {
    ext {
        buildToolsVersion = "32.0.0"
        minSdkVersion = 21
        compileSdkVersion = 33
        targetSdkVersion = 32
        androidXBrowser = "1.4.0"
        
        kotlinVersion = "1.8.0"
    }
...
}

```

**Comment by taltrui** [comment]
For those who setting `kotlinVersion = "1.8.0"` in `build.gradle` doesn't fix the issue, you can set `androidXVersion = "1.5.+"` instead. 

If you look in the `inappbrowser` `build.gradle`:

```
def androidXVersion = safeExtGet('androidXVersion', null)
  if (supportLibVersion && androidXVersion == null) {
    implementation "com.android.support:support-annotations:$supportLibVersion"
    implementation "com.android.support:customtabs:$supportLibVersion"
  } else {
    def defaultAndroidXVersion = "1.+"
    if (androidXVersion == null) {
      androidXVersion = defaultAndroidXVersion
    }
```

`def androidXVersion = safeExtGet('androidXVersion', null)`
This line will get the `androidXVersion` defined in your `build.gradle` and use it instead of the default `"1.+"`.

**Comment by costaDZ** [comment]
Took for me two days of investigation , after adding  **kotlinVersion = "1.8.0"** to the build.gradle file it's working for me right now .
